src/main/java/co/cask/hydrator/plugin/RecordSplitter.java

```diff
@@ -133,6 +136,17 @@ public void validate() throws IllegalArgumentException {
       } catch (IOException e) {
         throw new IllegalArgumentException("Unable to parse output schema.", e);
       }
+      Schema inputFieldSchema = inputSchema.getField(fieldToSplit).getSchema();
+      if (inputFieldSchema.isSimpleOrNullableSimple()) {
+        Schema.Type inputFieldType = (inputFieldSchema.isNullableSimple())
+          ? inputFieldSchema.getNonNullable().getType()
+          : inputFieldSchema.getType();
+        if (!(inputFieldType == Schema.Type.STRING)) {
+          throw new IllegalArgumentException("Source field: " + fieldToSplit +
+                                             " must be of type string. It is type: " +
+                                             inputFieldType.name());
+        }
```

Comment:
This will not catch the if the schema is of complex type. Throw an exception in else{}